factory/src/lib.rs

```diff
@@ -0,0 +1,555 @@
+        factory.admin = ctx.accounts.controller_program.key();
+        factory.mint_request_count = 0;
+        factory.burn_request_count = 0;
+
+        Ok(())
+    }
+
+    pub fn set_custodian_deposit_address(ctx: Context<SetCustodianDepositAddress>, deposit_address: String) -> Result<()> {
+        let members_state = ctx.accounts.member_state.clone();
+
+        if !is_merchant(ctx.accounts.merchant.key(), ctx.accounts.merchant_state.clone(), ctx.accounts.members.key(), ctx.accounts.member_state.key().clone()) {
+            return Err(Errors::SenderNotAuthorized.into())
+        }
+
+        if members_state.custodian == ctx.accounts.signer.key() {
+            return Err(Errors::SenderNotAuthorized.into())
+        }
+
+        if deposit_address.len() == 0 {
+            return Err(Errors::InvalidDepositAddress.into())
+        }
+        
+        let custodian_deposit_address = &mut ctx.accounts.custodian_deposit_address;
+        custodian_deposit_address.address = deposit_address;
+        custodian_deposit_address.bump = *ctx.bumps.get("custodian_deposit_address").unwrap();
+
+        Ok(())
+    }
+
+    pub fn set_merchant_deposit_address(ctx: Context<SetMerchantDepositAddress>, deposit_address: String) -> Result<()> {
+        if !is_merchant(ctx.accounts.merchant.key(), ctx.accounts.merchant_state.clone(), ctx.accounts.members.key(), ctx.accounts.member_state.key().clone()) {
+            return Err(Errors::SenderNotAuthorized.into())
+        }
+
+        if deposit_address.len() == 0 {
+            return Err(Errors::InvalidDepositAddress.into())
+        }
+        
+        let merchant_deposit_address = &mut ctx.accounts.merchant_deposit_address;
+        merchant_deposit_address.address = deposit_address;
+        merchant_deposit_address.bump = *ctx.bumps.get("merchant_deposit_address").unwrap();
+
+        Ok(())
+    }
+
+    pub fn add_mint_request(ctx: Context<AddMintRequest>, txid: String, deposit_address: String, amount: u64) -> Result<()> {
+        let factory_state = &mut ctx.accounts.factory_state;
+
+        if !is_merchant(ctx.accounts.merchant.key(), ctx.accounts.merchant_state.clone(), ctx.accounts.members.key(), ctx.accounts.member_state.key().clone()) {
+            return Err(Errors::SenderNotAuthorized.into())
+        }
+
+        if deposit_address != ctx.accounts.custodian_deposit_address.address.to_string() {
+            return Err(Errors::InvalidDepositAddress.into())
+        }
+
+        if txid.len() == 0 {
+            return Err(Errors::InvalidTxid.into())
+        }
+
+        let timestamp: u64 = clock::Clock::get().unwrap().unix_timestamp.try_into().unwrap();
+
+        let mint_request = &mut ctx.accounts.request;
+        mint_request.requester = ctx.accounts.merchant.key();
+        mint_request.amount = amount;
+        mint_request.deposit_address = deposit_address;
+        mint_request.txid = txid;
+        mint_request.nonce = factory_state.mint_request_count + 1;
```

Comment:
this will never overflow in our lifetime, but it is better to be consistent and use checked math everywhere